plasmapy/dispersion/numerical/stix_.py

```diff
@@ -0,0 +1,257 @@
+            " ion. The following particle(s) is(are) not allowed "
+            f"{[ion for ion, fail in zip(ions, failed) if not fail]}"
+        )
+
+    # Validate n_i argument
+    if n_i.ndim not in (0, 1):
+        raise ValueError(
+            "Argument 'n_i' must be a single valued or a 1D array of "
+            f"size 1 or {len(ions)}, instead got shape of {n_i.shape}"
+        )
+    elif n_i.ndim == 1 and n_i.size != len(ions):
+        raise ValueError(
+            "Argument 'n_i' and 'ions' need to be the same length, got"
+            f" value of shape {len(ions)} and {len(n_i.shape)}."
+        )
+
+    n_i = n_i.value
+    if n_i.ndim == 0:
+        n_i = np.array([n_i] * len(ions))
+    elif n_i.size == 1:
+        n_i = np.repeat(n_i, len(ions))
+
+    species = ions + [Particle("e-")]
+    densities = np.zeros(n_i.size + 1)
+    densities[:-1] = n_i
+    densities[-1] = np.sum(n_i * ions.charge_number)
+
+    # Validate B argument
+    B = B.squeeze()
+    if B.ndim != 0:
+        raise ValueError(
+            "Argument 'B' must be single valued and not an array of"
+            f" shape  {B.shape}."
+        )
+
+    # Validate w argument and dimension
+    w = w.squeeze()
+    if not (w.ndim == 0 or w.ndim == 1):
+        raise ValueError(
+            f"Argument 'w' needs to be a single value or a 1D array "
+            f" astropy Quantity, got a value of shape {w.shape}."
+        )
+    if np.any(w <= 0):
+        raise ValueError(f"Argument 'w' can not have a negative value.")
+    if np.isscalar(w.value):
+        w = np.array([w.value])
+
+    # Validate theta value
+    theta = theta.squeeze()
+    theta = theta.to(u.radian)
+    if theta.ndim not in (0, 1):
+        raise TypeError(
+            f"Argument 'theta' needs to be a single value or 1D array "
+            f" astropy Quantity, got array of shape {theta.shape}."
+        )
+    if np.isscalar(theta.value):
+        theta = np.array([theta.value])
+
+    # Generate mesh grid of w x theta
+    w, theta = np.meshgrid(w, theta, indexing="ij")
+
+    # Generate the plasma parameters needed
+    wps = []
+    wcs = []
+
+    for par, dens in zip(species, densities.tolist()):
+        wps.append(plasma_frequency(n=dens * u.m ** -3, particle=par).value)
+        wcs.append(gyrofrequency(B=B, particle=par, signed=False).value)
+    wps = np.array(wps)
+    wcs = np.array(wcs)
+
+    # Stix method implemented
+    S = 1
+    P = 1
+    D = 0
+    for wc, wp in zip(wcs, wps):
+        S -= (wp ** 2) / (w ** 2 - wc ** 2)
+        P -= (wp / w) ** 2
+        D += ((wp ** 2) / (w ** 2 - wc ** 2)) * (wc / w)
+
+    R = S + D
+    L = S - D
+
+    # Generate coefficients to solve
+    A = (c_si_unitless / w) ** 4 * (S * (np.sin(theta) ** 2) + P * (np.cos(theta) ** 2))
+    B = -((c_si_unitless / w) ** 2) * (
+        R * L * (np.sin(theta) ** 2) + P * S * (1 + (np.cos(theta) ** 2))
+    )
+    C = P * R * L
```

Comment:
```suggestion
    # Generate coefficients to solve, a * k**4 + b * k**2 + c = 0
    a = (c_si_unitless / w) ** 4 * (S * (np.sin(theta) ** 2) + P * (np.cos(theta) ** 2))
    b = -((c_si_unitless / w) ** 2) * (
        R * L * (np.sin(theta) ** 2) + P * S * (1 + (np.cos(theta) ** 2))
    )
    c = P * R * L
```

Make these variables lowercase to be consistent with PEP8 standards.  I don't make this same comment with S, P, D, R, and L because those variables names have a history in existing literature and that is overriding the PEP8 rules.

Obviously these changes will need to be reflected below when solving for `k`.